### PR TITLE
Fix some vampire things

### PR DESCRIPTION
**Pull request by ShiftyRail**
I don't know how entropy made it so these changes happened because I'm perfectly sure they worked at some point in time.
Anyway, this fixes null rods not providing adequate protection to their wielders against vampire powers as well as some chaplain immunities that were not working.

:cl:
- bugfix: Null rods are now properly blocking vampire powers again.

### CODE REVIEW

File: code/modules/spells/vampire_helpers.dm [hunk 1]
```diff
@@ -89,10 +89,10 @@
 		var/datum/role/vampire/V = M.GetRole(VAMPIRE)
 		var/obj/item/weapon/nullrod/N = locate(/obj/item/weapon/nullrod) in get_contents_in_object(src)
 		if (N)
-			if (/datum/power/vampire/undying in V.current_powers)
+			if (locate(/datum/power/vampire/undying) in V.current_powers)
 				to_chat(M.current, "<span class='warning'>An holy artifact has turned our powers against us!</span>")
 				return VAMP_FAILURE
-			if (/datum/power/vampire/jaunt in V.current_powers)
+			if (locate(/datum/power/vampire/jaunt) in V.current_powers)
 				to_chat(M.current, "<span class='warning'>An holy artifact protects [src]!</span>")
```
Comment (boy2mantwicethefam) [inline code/modules/spells/vampire_helpers.dm:96]: *A holy

File: code/modules/spells/vampire_helpers.dm [hunk 2]
```diff
@@ -89,10 +89,10 @@
 		var/datum/role/vampire/V = M.GetRole(VAMPIRE)
 		var/obj/item/weapon/nullrod/N = locate(/obj/item/weapon/nullrod) in get_contents_in_object(src)
 		if (N)
-			if (/datum/power/vampire/undying in V.current_powers)
+			if (locate(/datum/power/vampire/undying) in V.current_powers)
 				to_chat(M.current, "<span class='warning'>An holy artifact has turned our powers against us!</span>")
```
Comment (boy2mantwicethefam) [inline code/modules/spells/vampire_helpers.dm:93]: *A holy

File: code/modules/spells/vampire_helpers.dm (hunk 1 above)
Comment (Eneocho) [reply code/modules/spells/vampire_helpers.dm:96]: Not necessarily. In British English, words starting with "h" are treated as if they had a vowel instead. While weird to US English speakers, it's not grammatically wrong.

Also, OOC: Thanks, now I know my target (shifty) is bri'ish

Comment (ShiftyRail) [reply code/modules/spells/vampire_helpers.dm:96]: An 'oly

**Comment by drawsstuff** [comment]
Vampires also have lost their x-ray / thermals for some reason. No one's bug reported it so far, so I'll leave a heads up here. You normally get them upon maturing. That and thralling implanted personel is broken as well, likely because of diety's cult pr.

**Comment by APCUnlocked** [comment]
> Vampires also have lost their x-ray / thermals for some reason. No one's bug reported it so far, so I'll leave a heads up here. You normally get them upon maturing. That and thralling implanted personel is broken as well, likely because of diety's cult pr.

There's a roughly 20+ days issue up about the x-ray/thermals bug already. It's really annoying.

**Comment by TGFH** [comment]
> 
> 
> Vampires also have lost their x-ray / thermals for some reason. No one's bug reported it so far, so I'll leave a heads up here. You normally get them upon maturing. That and thralling implanted personel is broken as well, likely because of diety's cult pr.

Feature.

**Comment by ShiftyRail** [comment]
Will fix the rest probably

**Pull request merged**